src/components/QuranReader/hooks/useFetchPageVerses.ts

```diff
@@ -0,0 +1,46 @@
+import useTranslation from 'next-translate/useTranslation';
+import { shallowEqual, useSelector } from 'react-redux';
+import useSWRImmutable from 'swr/immutable';
+
+import { getPageVerses } from '@/api';
+import useGetMushaf from '@/hooks/useGetMushaf';
+import { selectIsUsingDefaultFont, selectQuranFont } from '@/redux/slices/QuranReader/styles';
+import { VersesResponse } from '@/types/ApiResponses';
+import { getDefaultWordFields } from '@/utils/api';
+import { makePageVersesUrl } from '@/utils/apiPaths';
+
+/**
+ * This hooks fetches the page's verses data by the pageId
+ *
+ * @param {string} pageId
+ * @param {VersesResponse} initialData
+ *
+ * @returns {{ pageVersesResponse: VersesResponse; isLoading: boolean; error: any }}
+ */
+
+const useFetchPageVerses = (pageId: string, initialData: VersesResponse) => {
+  const { lang: locale } = useTranslation();
+  const quranFont = useSelector(selectQuranFont, shallowEqual);
+  const isUsingDefaultFont = useSelector(selectIsUsingDefaultFont);
+  const mushafId = useGetMushaf();
+
+  const params = {
+    perPage: 'all',
+    mushaf: mushafId,
+    filterPageWords: true,
+    ...getDefaultWordFields(quranFont),
+  };
```

Comment:
Since this is common with `getStaticProps` inside `src/pages/page/[pageId].tsx`, we should create a common util function between both.

Reply:
Makes sense.